app/src/main/java/com/android/ubclaunchpad/driver/UI/DestinationActivity.java

```diff
@@ -98,116 +98,118 @@ protected void onCreate(Bundle savedInstanceState) {
+                    if (user != null && FirebaseUtils.getFirebaseUser() != null) {
+                        // move onto the next screen if both the inputs are filled in when button is clicked
+                        if (mCurrLoc != null && mDestLoc != null) {
+                            // update user information on user manager
+                            user.setCurrentLatLngStr(StringUtils.latLngToString(mCurrLoc));
+                            user.setDestinationLatLngStr(StringUtils.latLngToString(mDestLoc));
+
+                            // get Firebase user and save the location information
+                            String uid = FirebaseUtils.getFirebaseUser().getUid();
+                            Log.d(TAG, "got uid: " + uid);
+                            // save to current location endpoint
+                            mDatabase.child(StringUtils.FirebaseUserEndpoint)
+                                    .child(uid)
+                                    .child(StringUtils.FirebaseCurrentLatLngStr)
+                                    .setValue(StringUtils.latLngToString(mCurrLoc));
+                            // save to destination endpoint
+                            mDatabase.child(StringUtils.FirebaseUserEndpoint)
+                                    .child(uid)
+                                    .child(StringUtils.FirebaseDestinationLatLngEndpoint)
+                                    .setValue(StringUtils.latLngToString(mDestLoc));
+                            goToMainActivity(v);
                         }
-                    });
-                    AlertDialog alert = builder.create();
-                    alert.show();
-
 
+                        // if not, give alert to remind user to include the location information
+                        else {
+                            AlertDialog.Builder builder = new AlertDialog.Builder(v.getContext());
+                            builder.setTitle("Empty Fields Detected");
+                            builder.setMessage("Please make sure you have indicated where you are " +
+                                    "and where you want to go.");
+                            builder.setCancelable(false);
+                            builder.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialogInterface, int i) {
+                                    dialogInterface.dismiss();
+                                }
+                            });
+                            AlertDialog alert = builder.create();
+                            alert.show();
+                        }
+                    }else {
+                        // something went wrong with grabbing the user, go back to login
+                        FirebaseUtils.getFirebaseAuth().signOut();
```

Comment:
also make sure to null the user because your check was for both not being null and you are not sure which was the one that triggered it